src/main/java/org/prebid/server/bidder/improvedigital/ImprovedigitalBidder.java

```diff
@@ -67,6 +75,45 @@ public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request
         return Result.withValues(httpRequests);
     }
 
+    private ExtUser getAdditionalConsentProvidersUserExt(BidRequest request) {
+        if (request.getUser().getExt() == null) {
+            return null;
+        }
+        final ExtUser extUser = request.getUser().getExt();
+        final String consentedProviders = validateConsentedProvidersSettings(extUser.getConsentedProvidersSettings());
+
+        if (consentedProviders == null || StringUtils.isBlank(consentedProviders)) {
+            return null;
+        }
+
+        final String substringAfterTilde = StringUtils.substringAfterLast(consentedProviders, "~");
+        final Object[] arrayOfSplitString =
+                Arrays.stream(substringAfterTilde.split(REGEX_SPLIT_STRING_BY_DOT)).toArray();
+
+        final JsonNode consentProviderSettingJsonNode;
+
+        try {
+            consentProviderSettingJsonNode = customJsonNode(arrayOfSplitString);
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+
+        return mapper.fillExtension(extUser, consentProviderSettingJsonNode);
+    }
+
+    private JsonNode customJsonNode(Object[] arrayOfSplitString) {
```

Comment:
Do we really need Object[] here? String[] is enough, if so, on line 91 can be: `substringAfterTilde.split(REGEX_SPLIT_STRING_BY_DOT)`.
Also, don't forget to change variable to String[].

Reply:
Yeah you're right we can just use String[] and it's enough for us